builtin/docker/platform.go

```diff
@@ -141,43 +144,80 @@ func (p *Platform) Status(
 
 	log.Debug("querying docker for container health")
 
-	resources := []*sdk.StatusReport_Resource{{
-		Name: containerInfo.Name,
-	}}
+	// Find the declared resource that corresponds to our observed container
+	var containerDeclaredResource *sdk.DeclaredResource
+	for _, r := range declaredResources.Resources {
+		if r != nil && r.Name == "container" {
+			containerDeclaredResource = r
+		}
+	}
+
+	containerCreatedTime, err := time.Parse(time.RFC3339, containerInfo.Created)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to parse docker timestamp %q: %s", containerInfo.Created, err)
+	}
+
+	// Redact environment variables so we don't expose secrets on resources. Unmarshal/remarshaling to avoid mutating
+	// containerInfo
+	stateJson, err := json.Marshal(containerInfo)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to marshal container info to json: %s", err)
+	}
+	var containerInfoRedacted types.ContainerJSON
+	if err := json.Unmarshal(stateJson, &containerInfoRedacted); err != nil {
+		return nil, status.Errorf(codes.Internal, "failed to unmarshal container info from json: %s", err)
+	}
+	containerInfoRedacted.Config.Env = []string{}
+	stateJson, err = json.Marshal(containerInfoRedacted)
```

Comment:
Using docker's containerInfo as our stateJson is the most complete thing we could do, but it's also pretty verbose. I think that's OK, because I don't think we'd ever expose this whole thing in a UI outside of some super verbose output mode, in which case the user will probably want to see as much context as possible for debugging.

Reply:
I agree I think its okay